From 592308130579176f485e1bcd8128666e19f12297 Mon Sep 17 00:00:00 2001
From: Volker Krause <vkrause@kde.org>
Date: Sun, 17 May 2026 10:38:57 +0800
Subject: [PATCH 3/3] Consistently namespace ZXing version defines Avoids
 clashing with the defines in newer ZXing versions.

Also, simplify the format hint ifdefs a bit.
---
 src/scanner/config-prison-scanner.h.in | 8 ++++----
 src/scanner/format.cpp                 | 5 +++--
 src/scanner/videoscannerworker.cpp     | 6 +++---
 3 files changed, 10 insertions(+), 9 deletions(-)

diff --git a/src/scanner/config-prison-scanner.h.in b/src/scanner/config-prison-scanner.h.in
index fe3855e..447221e 100644
--- a/src/scanner/config-prison-scanner.h.in
+++ b/src/scanner/config-prison-scanner.h.in
@@ -6,9 +6,9 @@
 #ifndef CONFIG_PRISON_SCANNER_H
 #define CONFIG_PRISON_SCANNER_H
 
-#define ZXING_VERSION_MAJOR @ZXing_VERSION_MAJOR@
-#define ZXING_VERSION_MINOR @ZXing_VERSION_MINOR@
-#define ZXING_VERSION_PATCH @ZXing_VERSION_PATCH@
-#define ZXING_VERSION ((@ZXing_VERSION_MAJOR@<<16)|(@ZXing_VERSION_MINOR@<<8)|(@ZXing_VERSION_PATCH@))
+#define KZXING_VERSION_MAJOR @ZXing_VERSION_MAJOR@
+#define KZXING_VERSION_MINOR @ZXing_VERSION_MINOR@
+#define KZXING_VERSION_PATCH @ZXing_VERSION_PATCH@
+#define KZXING_VERSION ((@ZXing_VERSION_MAJOR@<<16)|(@ZXing_VERSION_MINOR@<<8)|(@ZXing_VERSION_PATCH@))
 
 #endif // CONFIG_PRISON_SCANNER_H
diff --git a/src/scanner/format.cpp b/src/scanner/format.cpp
index b7e8914..8536a88 100644
--- a/src/scanner/format.cpp
+++ b/src/scanner/format.cpp
@@ -4,6 +4,7 @@
 */
 
 #include "format_p.h"
+#include "config-prison-scanner.h"
 
 using namespace Prison;
 
@@ -34,7 +35,7 @@ static constexpr const format_map_t format_map[] = {
 
 ZXing::BarcodeFormats Format::toZXing(Format::BarcodeFormats formats)
 {
-#if ZXING_VERSION < QT_VERSION_CHECK(3, 0, 0)
+#if KZXING_VERSION < QT_VERSION_CHECK(3, 0, 0)
     ZXing::BarcodeFormats f;
 #else
     std::vector<ZXing::BarcodeFormat> f;
@@ -42,7 +43,7 @@ ZXing::BarcodeFormats Format::toZXing(Format::BarcodeFormats formats)
 
     for (auto m : format_map) {
         if (m.format & formats) {
-#if ZXING_VERSION < QT_VERSION_CHECK(3, 0, 0)
+#if KZXING_VERSION < QT_VERSION_CHECK(3, 0, 0)
             f |= m.zxFormat;
 #else
             f.push_back(m.zxFormat);
diff --git a/src/scanner/videoscannerworker.cpp b/src/scanner/videoscannerworker.cpp
index 3fa3464..41b70ac 100644
--- a/src/scanner/videoscannerworker.cpp
+++ b/src/scanner/videoscannerworker.cpp
@@ -26,12 +26,12 @@ VideoScannerWorker::VideoScannerWorker(QObject *parent)
 
 void VideoScannerWorker::slotScanFrame(VideoScannerFrame frame)
 {
-#if ZXING_VERSION < QT_VERSION_CHECK(2, 3, 0)
+#if KZXING_VERSION < QT_VERSION_CHECK(2, 3, 0)
     ZXing::Result zxRes;
 #else
     ZXing::Barcode zxRes;
 #endif
-#if ZXING_VERSION < QT_VERSION_CHECK(2, 3, 0)
+#if KZXING_VERSION < QT_VERSION_CHECK(2, 3, 0)
     ZXing::DecodeHints hints;
 #else
     ZXing::ReaderOptions hints;
@@ -124,7 +124,7 @@ void VideoScannerWorker::slotScanFrame(VideoScannerFrame frame)
     if (zxRes.isValid()) {
         auto res = ScanResultPrivate::get(scanResult);
 
-#if ZXING_VERSION < QT_VERSION_CHECK(1, 4, 0)
+#if KZXING_VERSION < QT_VERSION_CHECK(1, 4, 0)
         // distinguish between binary and text content
         const auto hasWideChars = std::any_of(zxRes.text().begin(), zxRes.text().end(), [](auto c) {
             return c > 255;
-- 
2.52.0

